Src/Web/Web.Shared.Net/RequestTrackingTelemetryModule.cs

```diff
@@ -118,7 +130,9 @@ public void AddTargetHashForResponseHeader(HttpContext context)
                 this.telemetryClient.Initialize(requestTelemetry);
             }
 
-            if (!string.IsNullOrEmpty(requestTelemetry.Context.InstrumentationKey) && context.Response.Headers[RequestResponseHeaders.TargetInstrumentationKeyHeader] == null)
+            if (ComponentCorrelation.CollectionEnabled
```

Comment:
This flag should disable subscribing to the callback completely, not just setting the header.